### PR TITLE
feature/CP-7017-ios-sdk-implement-initialize-listener

### PR DESCRIPTION
**Pull request by cpandya25**
implemented an initialise callback listener.

### CODE REVIEW

File: CleverPush/Source/CleverPushInstance.m [hunk 1]
```diff
@@ -254,16 +255,26 @@ - (id)initWithLaunchOptions:(NSDictionary*)launchOptions channelId:(NSString*)ne
...
+    return [self initWithLaunchOptions:launchOptions channelId:newChannelId handleNotificationReceived:NULL handleNotificationOpened:openedCallback handleSubscribed:subscribedCallback autoRegister:autoRegisterParam initialized:NULL];
+}
+
+#pragma mark - Common function to initialize SDK and sync data to NSUserDefaults
+- (id)initWithLaunchOptions:(NSDictionary*)launchOptions
+                  channelId:(NSString*)newChannelId
+ handleNotificationReceived:(CPHandleNotificationReceivedBlock)receivedCallback
+   handleNotificationOpened:(CPHandleNotificationOpenedBlock)openedCallback
+           handleSubscribed:(CPHandleSubscribedBlock)subscribedCallback
+               autoRegister:(BOOL)autoRegisterParam initialized:(CPInitializedBlock)initializedCallback {
```
Comment (gbhrdt) [inline CleverPush/Source/CleverPushInstance.m:273]: Please rename `initialized:(CPInitializedBlock)initializedCallback` to `handleInitialized:(CPInitializedBlock)initializedCallback` and put it into a new line so each parameter has 1 line

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CleverPushInstance.m [hunk 2]
```diff
@@ -402,6 +414,9 @@ - (void)initWithChannelId {
                 if (handleSubscribedInternal) {
                     handleSubscribedInternal(subscriptionId);
                 }
+                if (handleInitialized) {
```
Comment (gbhrdt) [inline CleverPush/Source/CleverPushInstance.m:417]: This needs to be moved, right now we only call handleInitialized when subscriptionId is non-nil which is not what we want.
We want to inform the customer if the init has been successful or not. So also on error we need to call handleInitialized(NO) and provide customer with the error there. For example if `getChannelConfig fails in the initialisation process.

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: CleverPush/Source/CleverPushInstance.m (hunk 2 above)
Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m]: In `getChannelConfing`, we have two main functions.

1. `[self getChannelConfigFromChannelId:configPath];`
2. `[self getChannelConfigFromBundleId:configPath];`

So I think that we need to handle success and failure as blocks in both functions.

That's why I keep the callback success or error message with the error message as well.
If it is successful, the error message will be null.
If it is false, the error message will be the same as what we are displaying in the log.

Comment (cpandya25) [reply CleverPush/Source/CleverPushInstance.m]: tested with the below scenarios (if I set the init callback function while using the init method).

1. First, I don't set the channelID, and we are getting false in the callback of the init listener.
2. Second, I set the channel ID, and we are getting success in the callback of the init listener.

**Pull request merged**